ludwig/utils/backward_compatibility.py

```diff
@@ -153,6 +154,49 @@ def _upgrade_trainer(trainer: Dict[str, Any]):
         trainer[EVAL_BATCH_SIZE] = None
 
 
+def _upgrade_preprocessing(preprocessing: Dict[str, Any]):
```

Comment:
Could we add at least one unit test for this file/function?

Usefully, this would also serve as a form of documentation we can link to that shows off what the deprecated format is and what the new format should be.